django/db/models/sql/query.py

```diff
@@ -1265,9 +1265,7 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
             condition = filter_expr.resolve_expression(self, allow_joins=allow_joins)
             if not isinstance(condition, Lookup):
                 condition = self.build_lookup(['exact'], condition, True)
-            clause = self.where_class()
-            clause.add(condition, AND)
-            return clause, []
+            return self.where_class([condition], connector=AND), []
```

Comment:
We can do the same for `reffed_expression`:
```diff
diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index b3d92d786c..21bc0aea7a 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1286,11 +1286,9 @@ class Query(BaseExpression):
         if check_filterable:
             self.check_filterable(value)
 
-        clause = self.where_class()
         if reffed_expression:
             condition = self.build_lookup(lookups, reffed_expression, value)
-            clause.add(condition, AND)
-            return clause, []
+            return self.where_class([condition], connector=AND), []
 
         opts = self.get_meta()
         alias = self.get_initial_alias()
@@ -1333,7 +1331,7 @@ class Query(BaseExpression):
 
         condition = self.build_lookup(lookups, col, value)
         lookup_type = condition.lookup_name
-        clause.add(condition, AND)
+        clause = self.where_class([condition], connector=AND)
 
         require_outer = lookup_type == 'isnull' and condition.rhs is True and not current_negated
         if current_negated and (lookup_type != 'isnull' or condition.rhs is False) and condition.rhs is not None:

```

Reply:
One day I'll correctly guess the level of invasiveness to apply, first time. I skipped this one precisely because it was added to multiple times subsequently and thought you might want the least disruption :D
I'll do this and another quick sweep/scan for any similarly distanced/return early ones.

Reply:
> I'll do this and another quick sweep/scan for any similarly distanced/return early ones.

I think that's the last one.

Reply:
Yeah, looks like all the rest are either constructions within a loop (which couldn't sensibly be converted to a listcomp/genexp) or conditionally used in more compicated ways, or adding to the `query.where/self.where` directly.